x-pack/plugins/transform/public/app/hooks/use_data_search.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { lastValueFrom } from 'rxjs';
+
+import type { IKibanaSearchRequest } from '@kbn/data-plugin/common';
+
+import { useAppDependencies } from '../app_dependencies';
+
+export const useDataSearch = () => {
+  const { data } = useAppDependencies();
+
+  return useCallback(
```

Comment:
did you consider moving the `AbortController` initialisation and calling `.abort` into this hook?

Reply:
I played around with it a bit, but it would then not be part of the consuming code's React lifecycle without an additional exposed API, e.g. within a `useEffect()` return function.